Core/DetectorConstruction.cpp

```diff
@@ -879,6 +985,280 @@ void DetectorConstruction::ConstructTargetRun7()
+ */
+
+void DetectorConstruction::ConstructTargetRun12()
+{
+  const double cyd_radius_inner = 12.0 * cm;
+  const double cyd_radius_outer = 12.3 * cm;
+  const double cyd_endcap_min = 12.7 * cm;
+  const double cyd_endcap_max = 14.0 * cm;
+  
+  G4double z_cyd[] = { -20.8 *cm,  20.8 * cm};
+  G4double rInner_cyd[] = { cyd_radius_inner, cyd_radius_inner };
+  G4double rOuter_cyd[] = { cyd_radius_outer, cyd_radius_outer};
+
+  G4Polycone* cydChamber = new G4Polycone(
+    "cydChamber", 0 * degree, 360 * degree, 2,
+    z_cyd, rInner_cyd, rOuter_cyd);
+
+  G4LogicalVolume* cydChamber_logical = new G4LogicalVolume(
+    cydChamber, fSiliconDioxide, "cydChamber_logical");
+
+  G4VisAttributes* detVisAtt = new G4VisAttributes(
+    G4Colour(0.8, 0.8, 0.0)
+  ); 
+  detVisAtt->SetForceWireframe(true);
+  detVisAtt->SetForceSolid(false);
+  cydChamber_logical->SetVisAttributes(detVisAtt);
+  
+  G4RotationMatrix rot = G4RotationMatrix();
+  G4ThreeVector loc = DetectorConstants::GetChamberCenter();
+  G4Transform3D transform(rot, loc);
+
+  new G4PVPlacement(
+     transform, cydChamber_logical, "cydChamberGeom",
+     fWorldLogical, true, 0, checkOverlaps);
+
+  G4double z_cydVacuum[] = { -20. *cm,  20. * cm };
+  G4double rOuter_cydVacuum[] = { cyd_radius_inner, cyd_radius_inner };
+  G4double rInner_cydVacuum[] = { 0 * cm, 0* cm};  
+
+  G4Polycone* cydChamberVacuum = new G4Polycone(
+    "cydChamberVacuum", 0 * degree, 360 * degree, 2,
+    z_cydVacuum, rInner_cydVacuum, rOuter_cydVacuum);
+
+  G4LogicalVolume* cydChamberVac_logical = new G4LogicalVolume(
+    cydChamberVacuum, fVacuum, "cydChamberVac_logical");
+
+  new G4PVPlacement( 
+    transform, cydChamberVac_logical, "cydChamberVac", 
+    cydChamber_logical, true, 0, checkOverlaps);
+
+  //  End Caps 
+  G4double z_endCap[] = {
+    -21.9 *cm, -21.8 *cm, -20.2 *cm, -20.1 *cm }; 
+  G4double rInner_endCap[] = {
+    0. *cm,  0. *cm, 0. *cm,  0. *cm};
+  G4double rOuter_endCap[] = {
+    cyd_endcap_min, cyd_endcap_max,  cyd_endcap_max, cyd_radius_outer};
+
+  G4Polycone* endCap = new G4Polycone(
+    "endCap", 0 * degree, 360 * degree, 4,
+    z_endCap, rInner_endCap, rOuter_endCap);
+
+  G4LogicalVolume* endCap_logical = new G4LogicalVolume(
+    endCap, fStainlessSteel, "endCap_logical");
+
+  G4VisAttributes* endCapVisAtt = new G4VisAttributes(
+    G4Colour(0.2, 0.4, 0.0)
+  );    
+  endCapVisAtt->SetForceWireframe(true);
+  endCapVisAtt->SetForceSolid(false);
+  endCap_logical->SetVisAttributes(endCapVisAtt);
+
+  new G4PVPlacement( 
+    transform, endCap_logical, "endCapGeom", 
+    fWorldLogical, true, 0, checkOverlaps);
+
+  G4RotationMatrix rot_endCap = G4RotationMatrix();
+  rot_endCap.rotateY(180 * degree);
+  G4Transform3D transform_endCap(rot_endCap, loc);
+  
+  new G4PVPlacement( 
+    transform_endCap, endCap_logical, "endCapGeom", 
+    fWorldLogical, true, 0, checkOverlaps);
+
+  // SphericalChamber
+  const double kapton_foil_radius_outer = 24 * mm;
```

Comment:
you did not used these variable. Is it the same as source_holder_radius_inner variable that you use in line 1226?